main.py

```diff
@@ -66,6 +71,27 @@ def run_data_checks_and_display_prompts(df):
             return False
     return True
 
+# Store that the form has not been submitted yet so we know to display the email form
+if 'form_submitted' not in st.session_state:
+    st.session_state['form_submitted'] = False
+
+# If the user has not submitted the form yet and an analytics key is set, display the email form
+if not st.session_state['form_submitted'] and segment_write_key is not None:
+    with st.form("email_form"):
+        st.write("Stay up to date with new Mito for Streamlit features by signing up for receive product updates.")
+        email = st.text_input("Email")
+
+        # Every form must have a submit button.
+        
+        submitted = st.form_submit_button("Submit")
+        if submitted:
+            # Send the email to segment
+            analytics.identify(email, {'location': 'streamlit_data_cleaning_verification_demo'})
+
+            # Store that the form has been submitted so we don't display it again
+            st.session_state['form_submitted'] = True
```

Comment:
This happens on the next rerender, so it requires pressing submit twice which isn't super nice.